tests/new_unit_tests/Commands/Connection/InfoCommandTest.php

```diff
@@ -0,0 +1,206 @@
+        $this->assertContains('redis_command', $parameters);
+    }
+
+    /**
+     * Exercises connection:info command with a valid environment and filter
+     *
+     * @return void
+     *
+     * @vcr site_connection-info
+     */
+    public function testConnectionInfoFilter()
+    {
+        // command output using a filter argument
+        $out        = $this->command->connectionInfo('behat-tests.dev', 'git_command');
+        $parameters = array_column($out->getArrayCopy(), 'param');
+
+        // assert parameter count
+        $this->assertCount(1, $parameters);
+
+        // assert filtered field
+        $this->assertContains('git_command', $parameters);
+    }
+
+    /**
+     * Exercises connection:info command without a valid environment argument
+     *
+     * @return void
+     */
+    public function testConnectionInfoInvalid()
+    {
+        $this->markTestIncomplete(
+            'Skipping testConnectionInfoInvalid() -- Needs work'
+        );
+
+        $this->setInput([
+            'command' => 'connection:info',
+            // TODO: This does not appear to be the correct way to specify the command arguments
+            ['invalid-env'],
+        ]);
+
+        // should display an error message
+        $this->assertEquals(
+            '[error]  The environment argument must be given as <site_name>.<environment>',
+            $this->runCommand()->fetchTrimmedOutput()
+        );
+        $this->assertNull($this->command->connectionInfo('behat-tests'));
+    }
+
+    /**
+     * Example test using Prophecy API
+     *
+     * @return void
+     */
+    public function testEnvironmentParamsProphecy()
+    {
+        $site_prophet = $this->prophet->prophesize(Site::class);
+        $site_prophet->get('name')->willReturn('my_site');
+        $site = $site_prophet->reveal();
+
+        $env_prophet = $this->prophet->prophesize(Environment::class);
+        $env_prophet->connectionInfo()->willReturn([
+            'param_a' => 'value_a',
+            'param_b' => 'value_b',
+        ]);
+        $environment       = $env_prophet->reveal();
+        $environment->id   = 'my_env';
+        $environment->site = $site;
+
+        // should return all parameters
+        $this->assertEquals(
+            [
+                ['env' => 'my_site.my_env', 'param' => 'param_a', 'value' => 'value_a'],
+                ['env' => 'my_site.my_env', 'param' => 'param_b', 'value' => 'value_b'],
+            ],
+            $this->protectedMethodCall($this->command, 'environmentParams', [$environment])
+        );
+
+        // should return only filtered parameter
+        $this->assertEquals(
+            [
+                ['env' => 'my_site.my_env', 'param' => 'param_b', 'value' => 'value_b'],
+            ],
+            $this->protectedMethodCall(
+                $this->command,
+                'environmentParams',
+                [$environment, 'param_b']
+            )
+        );
+    }
+
```

Comment:
The test above and below exercise the same logic using different approaches to generating dummies.
Does anyone have opinions/experience with MockBuilder vs Prophecy approach to pick one over the other? They seem functionally equivalent, at least as far as this test goes, but I want to identify the prefered approach and delete the other.